lib/tests/testscripts/base_testscript.rb

```diff
@@ -154,21 +177,28 @@ def perform_action(action)
       def execute_operation(operation)
         return if @client.nil?
         requestHeaders = Hash[operation.requestHeader.all.map{|u| [u.field, u.value]}] #Client needs upgrade to support
+        format = FHIR::Formats::ResourceFormat::RESOURCE_XML
+        format = FORMAT_MAP[operation.contentType] unless operation.contentType.nil?
+        format = FORMAT_MAP[operation.accept] unless operation.accept.nil?
         case operation.fhirType.code
         when 'read'
           if !operation.targetId.nil?
             @last_response = @client.read @fixtures[operation.targetId].class, @id_map[operation.targetId]
           else
-            resource_type = operation.resource
-            resource_id = operation.params
+            resource_type = replace_variables(operation.resource)
+            resource_id = replace_variables(operation.params)
             @last_response = @client.read "FHIR::#{resource_type}".constantize, id_from_path(resource_id)
-            
           end
         when 'vread'
           raise 'vread not implemented'
         when 'search'
-          params = extract_operation_parameters(operation)
-          @last_response = @client.search "FHIR::#{operation.resource}".constantize, search: {parameters: params}
+          if operation.url.nil?
+            params = extract_operation_parameters(operation)
+            @last_response = @client.search "FHIR::#{operation.resource}".constantize, {search: {parameters: params}}, format
+          else
+            url = replace_variables(operation.url)
+            last_response = @client.search "FHIR::#{operation.resource}".constantize, url: url #todo implement URL
```

Comment:
Yeah, good catch.  I missed it in part because that line is defunct now, waiting for another fix (client accepting arbtrary URLs... yuck).  That's in the tracker.